app/demo-loop/RootOutput.cc

```diff
@@ -29,33 +31,21 @@ void write_to_root(RunnerInput const& cargs,
 {
     CELER_EXPECT(cargs);
 
-    auto& args = const_cast<RunnerInput&>(cargs);
-    auto tree_input = root_manager->make_tree("input", "input");
-
-    // Problem definition
-    tree_input->Branch("geometry_filename", &args.geometry_filename);
-    tree_input->Branch("physics_filename", &args.physics_filename);
-    tree_input->Branch("hepmc3_filename", &args.hepmc3_filename);
-
-    // Control
-    tree_input->Branch("seed", &args.seed);
-    tree_input->Branch("max_num_tracks", &args.max_num_tracks);
-    tree_input->Branch("max_steps", &args.max_steps);
-    tree_input->Branch("initializer_capacity", &args.initializer_capacity);
-    tree_input->Branch("max_events", &args.max_events);
-    tree_input->Branch("secondary_stack_factor", &args.secondary_stack_factor);
-    tree_input->Branch("enable_diagnostics", &args.enable_diagnostics);
-    tree_input->Branch("use_device", &args.use_device);
-    tree_input->Branch("sync", &args.sync);
-    tree_input->Branch("step_limiter", &args.step_limiter);
+    // Input
+    nlohmann::json json_inp;
+    to_json(json_inp, cargs);
+    std::string str_input(json_inp.dump());
```

Comment:
I think this should work:
```suggestion
    std::string str_input(nlohmann::json(cargs).dump());
```